src/lib/project-saver-hoc.jsx

```diff
@@ -54,14 +54,16 @@ const ProjectSaverHOC = function (WrappedComponent) {
             // if we're newly able to create this project on the server, create it!
             const showingCreateable = this.props.canCreateNew && this.props.isShowingWithoutId;
             const prevShowingCreateable = prevProps.canCreateNew && prevProps.isShowingWithoutId;
-            if (showingCreateable && !prevShowingCreateable) {
+            const justTriedCreatingNew = prevProps.isCreatingNew; // don't immediately keep trying
+            if (showingCreateable && !prevShowingCreateable && !justTriedCreatingNew) {
                 this.props.onCreateProject();
             } else {
                 // if we're newly able to save this project, save it!
                 const showingSaveable = this.props.canSave && this.props.isShowingWithId;
                 const becameAbleToSave = this.props.canSave && !prevProps.canSave;
                 const becameShared = this.props.isShared && !prevProps.isShared;
-                if (showingSaveable && (becameAbleToSave || becameShared)) {
+                const justTriedUpdating = prevProps.isUpdating; // don't immediately keep trying
+                if (showingSaveable && !justTriedUpdating && (becameAbleToSave || becameShared)) {
```

Comment:
This is getting pretty unwieldy. It's pretty hard to understand both of these conditions. Can we use a guard clause and just avoid getting here in case of the excluded conditions (`!prevShowingCreatable`, `!justTriedCreatingNew`, `!justTriedUpdating`)?

Reply:
("both of these conditions", meaning this if clause and the one that's been changed above.)

Reply:
I see what you mean. Take a look now, is this better?